test/e2e/generator/generate.go

```diff
@@ -273,7 +279,14 @@ func generateTestnet(r *rand.Rand, opt map[string]interface{}) (e2e.Manifest, er
 		if len(seedNames) > 0 && (i == 0 || r.Float64() >= 0.5) {
 			manifest.Nodes[name].Seeds = uniformSetChoice(seedNames).Choose(r)
 		} else if i > 0 {
-			manifest.Nodes[name].PersistentPeers = uniformSetChoice(peerNames[:i]).Choose(r)
+			ppeers := uniformSetChoice(peerNames[:i]).Choose(r)
```

Comment:
I assume this is on purposes, but do we know at this point that the first i elements of this slice are _only_ persistent peers? Or is it that we don't care and just need some peers to designate as persistent?

Reply:
I'm really not sure at all. I think it's "all the nodes before the current node, which are sorted based on what nodes need to do catch up." (so nodes that are later in the list need to catch up more?) 

I know that any node that doesn't have a seed or a persistent peer list has the entire network setup as persistent peers. I'll tweak this a bit to try and make sure we don't end up with insufficient numbers of peers at startup.